addons/html_editor/static/src/core/selection_plugin.js

```diff
@@ -479,10 +478,16 @@ export class SelectionPlugin extends Plugin {
             throw new Error("Selection is not in editor");
         }
         const isCollapsed = anchorNode === focusNode && anchorOffset === focusOffset;
-        [focusNode, focusOffset] = normalizeCursorPosition(focusNode, focusOffset, "right");
+        [focusNode, focusOffset] = normalizeSelfClosingElement(focusNode, focusOffset, "right");
         [anchorNode, anchorOffset] = isCollapsed
             ? [focusNode, focusOffset]
-            : normalizeCursorPosition(anchorNode, anchorOffset, "left");
+            : normalizeSelfClosingElement(anchorNode, anchorOffset, "left");
+        // In conformity to some spec we don't understand why (ask FGE!):
+        // "set a collapse selection in a contenteditable false should move it after this node"
+        if (isCollapsed) {
+            const [node, offset] = normalizeNotEditableNode(anchorNode, anchorOffset, "right");
+            [anchorNode, anchorOffset, focusNode, focusOffset] = [node, offset, node, offset];
+        }
```

Comment:
We should definitely get rid of this. Either the comment, either the whole `if` block.
@FrancoisGe Do you remember the reason for this? https://github.com/odoo/odoo/pull/181002

Reply:
To be honest, I don t remember. What appens with your fix ? It was probably a dirty fix ^^